hnn_core/lfp.py

```diff
@@ -0,0 +1,221 @@
+"""
+LFPsim - Simulation scripts to compute Local Field Potentials (LFP) from cable
+compartmental models of neurons and networks implemented in NEURON simulation
+environment.
+
+LFPsim works reliably on biophysically detailed multi-compartmental neurons
+with ion channels in some or all compartments.
+
+Last updated 12-March-2016
+Developed by :
+Harilal Parasuram & Shyam Diwakar
+Computational Neuroscience & Neurophysiology Lab, School of Biotechnology,
+Amrita University, India.
+
+Email: harilalp@am.amrita.edu; shyam@amrita.edu
+www.amrita.edu/compneuro
+
+translated to Python and modified to use use_fast_imem by Sam Neymotin
+based on mhines code
+
+References
+----------
+1 . Parasuram H, Nair B, D'Angelo E, Hines M, Naldi G, Diwakar S (2016)
+Computational Modeling of Single Neuron Extracellular Electric Potentials
+and Network Local Field Potentials using LFPsim. Front Comput Neurosci 10:65
+[PubMed].
+"""
+
+# Authors: Mainak Jas <mjas@mgh.harvard.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Sam Neymotin <samnemo@gmail.com>
+
+from neuron import h
+
+import numpy as np
+from numpy.linalg import norm
+
+
+def _get_all_sections(sec_type='Pyr'):
+    ls = h.allsec()
+    ls = [s for s in ls if sec_type in s.name()]
+    return ls
+
+
+def _get_segment_counts(all_sections):
+    seg_counts = np.zeros((len(all_sections), ), dtype=np.int)
+    for ii, sec in enumerate(all_sections):
+        seg_counts[ii] = np.int(sec.nseg)
```

Comment:
```suggestion
    seg_counts = [np.int(sec.nseg) for sec in all_sections]
```

don't see a reason you need a numpy array for this ... also weird that `np.int` is needed. Neuron gives floating point for `sec.nseg` ?

Reply:
My bad: `nseg` returns `int`. I think I had plans to use `seg_counts` differently, will simplify. Generally speaking, I think very carefully about using python lists. Sure, they're super-easy to work with. But as soon as we're talking about thousands (potentially hundreds of thousands) of elements, the overhead becomes massive. Our built `net` object is already heavy, so I'm trying to reduce the baggage.